src/sources/util/grpc/mod.rs

```diff
@@ -51,10 +56,48 @@ where
         .layer(DecompressionAndMetricsLayer)
         .add_service(service)
         .serve_with_incoming_shutdown(stream, shutdown.map(|token| tx.send(token).unwrap()))
-        .in_current_span()
         .await?;
 
     drop(rx.await);
 
     Ok(())
 }
+
+pub fn build_grpc_trace_layer(
+    span: Span,
+) -> TraceLayer<
+    SharedClassifier<GrpcErrorsAsFailures>,
+    impl Fn(&Request<Body>) -> Span + Clone,
+    impl Fn(&Request<Body>, &Span) + Clone,
+    impl Fn(&Response<BoxBody>, Duration, &Span) + Clone,
+    (),
+    (),
+    (),
+> {
+    TraceLayer::new_for_grpc()
+        .make_span_with(move |request: &Request<Body>| {
+            // The path is defined as “/” {service name} “/” {method name}.
+            let mut path = request.uri().path().split('/');
+            let service = path.nth(1).unwrap_or("_unknown");
+            let method = path.next().unwrap_or("_unknown");
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏just wondering if the expectation is for these to always be known or if there are legitimate cases where they might not be specified, in which case maybe "unspecified" might be more appropriate, as I think users would be less likely to see that as a problem. If we do expect it to be there it might be useful to add to the existing comment about the `path` , that they <should> be there.

Reply:
These _should_ never be unknown given that the gPRC protocol defines the request path structure.